lib/src/boilerplate_suggestors/boilerplate_utilities.dart

```diff
@@ -0,0 +1,121 @@
+// Copyright 2020 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:over_react_codemod/src/constants.dart';
+import 'package:over_react_codemod/src/util.dart';
+
+typedef void YieldPatch(
+    int startingOffset, int endingOffset, String replacement);
+
+// Stub while <https://jira.atl.workiva.net/browse/CPLAT-9308> is in progress
+bool _isPublic(ClassDeclaration node) => false;
+
+/// Whether a props or state class class [node] should be migrated as part of the boilerplate codemod.
+bool shouldMigratePropsAndStateClass(ClassDeclaration node) {
+  return isAssociatedWithComponent2(node) &&
+      isAPropsOrStateClass(node) &&
+      // Stub while <https://jira.atl.workiva.net/browse/CPLAT-9308> is in progress
+      !_isPublic(node);
+}
+
+/// A simple RegExp against the parent of the class to verify it is `UiProps`
+/// or `UiState`.
+bool extendsFromUiPropsOrUiState(ClassDeclaration classNode) =>
+    classNode.extendsClause.superclass.name
```

Comment:
This is also causing errors while testing on `web_skin_dart`. I think it just needs a null check.